### PR TITLE
feat: add first letters to transliterations

### LINKED ISSUE #1780: Add {english, hindi, etc.} translit first letters
Do you think it's maybe worth doing the english translit or hindi translit first letters in this PR?

_Originally posted by @bhajneet in https://github.com/ShabadOS/database/pull/1776#issuecomment-638467784_

### PR DESCRIPTION
**Pull request by Harjot1Singh**
### Summary of PR
Adds `first_letters` and `vishraam_first_letters` to transliterations.

Also includes a correct for the `first_letters` column of `gurmukhi`.

### Tests for unexpected behavior
- Went through data in the columns

### Time spent on PR
2 hours

### Linked issues
Fix #1780

### Reviewers
@bhajneet

### CODE REVIEW

File: lib/build-database.js [hunk 1]
```diff
@@ -321,23 +322,31 @@ const importLines = async trx => {
...
             const sourceId = sourceNames.findIndex( nameEnglish => sourceName === nameEnglish ) + 1
 
-            lineContentData.push( {
-              line_id: line.id,
-              source_id: sourceId,
-              gurmukhi,
+            const getFirstLetters = text => ( {
               first_letters: [
+                toUnicode,
                 stripEndings,
```
Comment (Sarabveer) [inline lib/build-database.js:328]: https://github.com/ShabadOS/gurmukhi-utils/pull/163

Check if line is `Sirlekh`

File: lib/build-database.js [hunk 2]
```diff
@@ -321,23 +322,31 @@ const importLines = async trx => {
...
                 stripAccents,
                 stripVishraams,
                 firstLetters,
-              ].reduce( ( text, fn ) => fn( text ), gurmukhi ),
+                toAscii,
+              ].reduce( ( text, fn ) => fn( text ), text ),
               vishraam_first_letters: [
+                toUnicode,
                 stripEndings,
```
Comment (Sarabveer) [inline lib/build-database.js:336]: ditto

**Comment by Harjot1Singh** [comment]
image

**Comment by Sarabveer** [comment]
LGTM

**Comment by bhajneet** [comment]
Is it correct to capture line-endings in first_letters?

**Comment by Harjot1Singh** [comment]
> Is it correct to capture line-endings in first_letters?

No, and we do not, except for Sirlekh lines (purely a limitation of our GU function)

**Pull request merged**